client/app/telecom/pack/xdsl/modem/templates/pack-xdsl-modem-templates.controller.js

```diff
@@ -0,0 +1,83 @@
+angular.module('managerApp').controller('XdslModemTemplateCtrl', class XdslModemTemplateCtrl {
+  /* @ngInject */
+  constructor($stateParams, $translate, $uibModal, OvhApiXdsl, TucPackXdslModemMediator,
+    TucToast) {
+    this.$stateParams = $stateParams;
+    this.$translate = $translate;
+    this.$uibModal = $uibModal;
+    this.mediator = TucPackXdslModemMediator;
+    this.modemTemplates = [];
+    this.OvhApiXdsl = OvhApiXdsl;
+    this.templateName = null;
+    this.TucToast = TucToast;
+  }
+
+  getModemTemplates() {
+    return this.OvhApiXdsl.TemplateModem().v6().query()
+      .$promise.then((templateModemIds) => {
+        this.modemTemplates = templateModemIds;
+        return templateModemIds;
+      }).catch(() => {
+        this.modemTemplates = [];
```

Comment:
yes I need to reset the list of templates if there is an error, to avoid the possibility to the user to select a template modem that is not available.